go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/palantir/conjure-go/v6
 
-go 1.21
+go 1.23.2
```

Comment:
If we want this to be 1.23, then this line should be `1.23.0`.

The idea is that we want it to be the first release of the major version so that the module can use the language features, but also want to be as permissive as possible for vendoring/use by other modules (we don't want to block import for a different module that declares `go 1.23.1`).

This line sets a *minimum* for the toolchain required to build it, but if the local go version is greater than this version, the module will still be built with the "newer" go toolchain (unless there is a different directive that forces a specific toolchain version)